sm64-tas-scripting/Inputs.cpp

```diff
@@ -329,57 +354,46 @@ int M64::load()
 	return 1;
 }
 
-bool Inputs::HauEquals(int16_t angle1, int16_t angle2)
-{
-	int16_t hau1 = angle1 - (angle1 & 15);
-	int16_t hau2 = angle2 - (angle2 & 15);
 
-	return hau1 == hau2;
-}
 
 int M64::save(long initFrame)
 {
-	FILE* f;
+	std::ofstream f(fileName, ios_base::binary);
+	f.exceptions(ios_base::failbit | ios_base::badbit);
 
 	uint16_t buttons;
 	int8_t stick_x, stick_y;
 
 	size_t err;
 	uint64_t lastFrame = frames.rbegin()->first;
 
-	try {
-		if ((err = fopen_s(&f, fileName, "r+b")) != 0) {
-			std::cerr << "Bad open of file " << fileName << " Error: " << err << std::endl;
-			exit(EXIT_FAILURE);
-		}
+	try
+	{
 
-		//Write number of frames
-		fseek(f, 0xC, SEEK_SET);
-		int32_t nFrames[1] = { -1 }; //To-Do: maybe change to size
-		fwrite(nFrames, sizeof(int32_t), 1, f);
-		if (ferror(f) != 0)
-			return 0;
+		// Write number of frames
+		f.seekp(0xC, ios_base::beg);
+		int32_t nFrames[1] = {-1};
+		f.write(reinterpret_cast<char*>(nFrames), sizeof(int32_t));
 
-		//Write frames
-		fseek(f, 0x400 + 4 * initFrame, SEEK_SET);
-		for (int i = 0; i <= lastFrame; i++) {
+		// Write frames
+		f.seekp(0x400 + 4 * initFrame, ios_base::beg);
+		for (int i = 0; i <= lastFrame; i++)
+		{
 			uint16_t bigEndianButtons = 0;
-			int8_t stickX = 0;
-			int8_t stickY = 0;
-			
+			int8_t stickX							= 0;
+			int8_t stickY							= 0;
+
 			if (frames.count(i))
 			{
-				bigEndianButtons = htons(frames[i].buttons);
-				stickX = frames[i].stick_x;
-				stickY = frames[i].stick_y;
+				bigEndianButtons = byteswap(frames[i].buttons);
+				stickX					 = frames[i].stick_x;
+				stickY					 = frames[i].stick_y;
 			}
-			
-			fwrite(&bigEndianButtons, sizeof(uint16_t), 1, f);
-			fwrite(&stickX, sizeof(int8_t), 1, f);
-			fwrite(&stickY, sizeof(int8_t), 1, f);
-		}
 
-		fclose(f);
+			f.write(reinterpret_cast<char*>(bigEndianButtons), sizeof(uint16_t));
```

Comment:
```suggestion
			f.write(reinterpret_cast<char*>(&bigEndianButtons), sizeof(uint16_t));
```

Reply:
This was fixed in the remove MTAP commit.

Reply:
I think it is still missing